loader/dd_library_loader.c

```diff
@@ -328,6 +334,16 @@ static PHP_MINIT_FUNCTION(ddloader_injected_extension_minit) {
         return SUCCESS;
     }
 
+    char *incompatible_exts[] = {"Xdebug", "the ionCube PHP Loader", "newrelic", "blackfire", "pcov"};
```

Comment:
They're not all incompatible? especially not xdebug. We even actively test for compatibility with it (tests/xdebug).

What I meant: add to https://github.com/DataDog/dd-trace-php/blob/master/ext/excluded_modules.c the extensions which are generally incompatible (ioncube).

Add here, extensions which have a risk to segfault when run alongside dd-trace. Afaik pcov, blackfire, xdebug and newrelic work nicely together.

Reply:
> They're not all incompatible? especially not xdebug

We have crash reports related to Xdebug, and we know that some versions contain bugs that can cause crashes. 
So IMHO, in the context of SSI, it's safer not to inject the tracer at all when any other extensions that might use the same internal hooks as we do are loaded, rather than disabling it.

That's why I made the changes in the loader instead of the tracer.

Reply:
We do check the compatibility before actually setting any hooks.